doc: add docs of github action usage (#4071)
bentoml/BentoML: docs/source/tutorial.rst

Learn more about different deployment options with BentoML from the
:doc:`concepts/deploy` page.

Using GitHub Actions to Boost Your Workflow
-------------------------------------------

BentoML provides a GitHub Action to help you automate the process of building Bentos and deploying them to the cloud.
To create a GitHub Actions workflow, you need to first define a workflow file as below:

.. literalinclude:: ./data/workflow.yaml
   :language: yaml

> You can get more details about this file in :ref:`Deploying Your Bento <bentocloud/how-tos/deploy-bentos:Deploying Your Bento>` and `the GitHub Actions documentation <https://docs.github.com/en/actions/quickstart>`_.

Reviewer: ```suggestion
The action [`bentoml/deploy-bento-action`](https://github.com/bentoml/deploy-bento-action) requires `cloud_api_token` and `cloud_endpoint`, which you can set the [repository secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) to make sure the job has correct privilege.

You can get more details about this file in :ref:`Deploying Your Bento <bentocloud/how-tos/deploy-bentos:Deploying Your Bento>` and `the GitHub Actions documentation <https://docs.github.com/en/actions/quickstart>`_.
```

Learn more about different deployment options with BentoML from the
:doc:`concepts/deploy` page.

> Use GitHub Action to Boost Your Workflow

Reviewer 2: ```suggestion
Using GitHub Actions to Boost Your Workflow
```

----------------------------------------

> BentoML provides a GitHub Action to help you automate the process of building bento and deploying to the cloud.

Reviewer 2: ```suggestion
BentoML provides a GitHub Action to help you automate the process of building Bentos and deploying them to the cloud.
```

> Here an example of the workflow file:

Reviewer 2: ```suggestion
To create a GitHub Actions workflow, you need to first define a workflow file as below:
```

> .. code-block:: yaml

Reviewer: I think we should make this a yaml file, and the link it here to rest for us to easier to maintain. As working in yaml file is better than working with yaml within rst

    name: Update Bento Deployment
    on:
    push:
      tags:
        - v*

    jobs:
      build_and_deploy:
        runs-on: ubuntu-latest
        steps:
          - uses: actions/checkout@v3

          - uses: actions/setup-python@v4
            with:
            python-version: '3.10'
            cache: 'pip'

          - name: Build and Deploy
            uses: bentoml/deploy-bento-action@main
              with:
                cloud_api_token: ${{ secrets.CLOUD_API_TOKEN }}
                cloud_endpoint: ${{ secrets.CLOUD_ENDPOINT }}
                deployment_name: test-iris

> You can get more details about this file in :ref:`Deploying Your Bento <bentocloud/how-tos/deploy-bentos:Deploying Your Bento>`.

Reviewer 2: ```suggestion
You can get more details about this file in :ref:`Deploying Your Bento <bentocloud/how-tos/deploy-bentos:Deploying Your Bento>` and `the GitHub Actions documentation <https://docs.github.com/en/actions/quickstart>`_.
```
